File: python/aoc_2022/day1/solution_a.py
if __name__ == "__main__":
    
    with open("inputs/day1/input_a.txt", "r") as f:
        input = f.read()
        
    groups = map(lambda group: map(lambda line: int(line), group), map(lambda group: group.split("\n"), input.split("\n\n")))    
    group_sums = map(lambda group: sum(group), groups)
    result = max(group_sums)
    print(result) # 69177

File: scala/src/main/scala/day1/SolutionA.scala
package day1

import java.nio.file.{Files, Paths}

object SolutionA extends App {

  val input = Files.readString(Paths.get("src/main/resources/day1/input_a.txt"))
  val groups = input.split("\n\n").map(_.split("\n").map(_.toInt))
  val mostCalories = groups.map(_.sum).max

  println(mostCalories) // 69177

}
